### ISSUE TITLE
MissingGhRepo false positive

### ISSUE THREAD

**Issue by TWiStErRob** [TWiStErRob/net.twisterrob.ghlint#280]
### Version

release after ghlint 0.5.0

### Reproduction steps

1. File `.github/workflows/x.yml` with the following content:
   ```yaml
   on: push
   jobs:
     repro:
       runs-on: ubuntu-latest
       steps:
         - run: gh pr list --repo=owner/repo
   ```

2. Run `ghlint` with the following arguments:
   ```shell
   ghlint --console .github/workflows/x.yml
   ```

### Expected behavior

No report, repository is explicit.

### Actual behavior

> Step[#0] in Job[repro]  should see `GH_REPO` environment variable or have a repository cloned.

### Additional Context

This is a known issue while implementing #247. We don't use this form yet, so the need didn't come up. Might be easier if/when the run steps are parsed: https://github.com/TWiStErRob/net.twisterrob.ghlint/issues/279
